lib/Futures/Future.cpp

```diff
@@ -21,7 +21,7 @@
 /// @author Simon Grätzer
 ////////////////////////////////////////////////////////////////////////////////
 
-#include "Futures/Future.h"
+#include "Future.h"
```

Comment:
revert this please

Reply:
has it ever occurred to you, that it might be completely obvious to you, but it isn't to others? Why do you want him to revert? It is not very respectful to just dump whatever you want. We are not your code monkeys. Please be better!

Reply:
I think if it obvious for Markus, he write to me why.
Sorry if it sounds rude. But for me, without explanation in PR description it looks like changes for changes

Reply:
The file `Future.h` is in the same directory as `Future.cpp` and can hence be included as such without a path. 

We can have a separate discussion about whether it is desirable for a library to not put its headers into a dedicated subdirectory; 
The library is declared in `CMakeLists.txt` to have just its own path as include directory.

That would mean having a subdirectory `lib/Futures/Futures` under `lib/Futures`; I found that unnecessarily verbose.

This becomes a problem once there is `lib/Lib1/Utils.h` and `lib/Lib2/Utils.h` and no way to disambiguate; so yes, sure, we can add the header directory.

Reply:
> The file Future.h is in the same directory as Future.cpp and can hence be included as such without a path.

Relative includes more slow on compilation stage.

But you don't use it, you use absolute include because you add it to include directories in cmake

> The library is declared in CMakeLists.txt to have just its own path as include directory.

But why? Maybe it a little safer from one side, but we don't develop library, it's just small part of arangodb, in separate directory, I don't understand the reason to do it.

> That would mean having a subdirectory lib/Futures/Futures under lib/Futures; I found that unnecessarily verbose.
This becomes a problem once there is lib/Lib1/Utils.h and lib/Lib2/Utils.h and no way to disambiguate; so yes, sure, we can add the header directory.

Yes, so you can just use include paths from "lib" not "lib/Futures" (as we do before for all other libs) and don't add additional stuff to cmake for compiler.
Also it protects us from situation that you describe

Reply:
I think overall your PR is good changes. And will be good to merge it, but maybe you can make separate PR with "include" changes, where we can discuss pros and cons?
As you write:

> We can have a separate discussion about whether it is desirable for a library to not put its headers into a dedicated subdirectory;